base/src/types.rs

```diff
@@ -106,8 +106,13 @@ pub enum Type<Id, T = AstType<Id>> {
     Record {
         /// The associated types of this record type
         types: Vec<Field<Id, Alias<Id, T>>>,
+        row: T,
+    },
+    EmptyRow,
```

Comment:
I think there is a `TRecord Type` variant in [`AlgorithmW_Records`](https://github.com/willtim/row-polymorphism/blob/master/src/AlgorithmW_Records.hs#L48) because you do need a record constructor...